controllers/node_controller.go

```diff
@@ -88,6 +93,27 @@ func (r *nodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (resul
 		return ctrl.Result{}, err
 	}
 
+	if _, ok := node.GetAnnotations()[metadata.RebootAnnotation]; ok {
+		// Create a new signer using the private key that the instances will be reconciled with
+		signer, err := signer.Create(types.NamespacedName{Namespace: r.watchNamespace,
+			Name: secrets.PrivateKeySecret}, r.client)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("unable to create signer from private key secret: %w", err)
+		}
+		instanceInfo, err := r.instanceFromNode(node)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		nc, err := nodeconfig.NewNodeConfig(r.client, r.k8sclientset, r.clusterServiceCIDR, r.watchNamespace,
+			instanceInfo, signer, nil, nil, r.platform)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to create new nodeconfig: %w", err)
+		}
+
+		if err := nc.SafeReboot(ctx, node); err != nil {
```

Comment:
I'm realizing that passing in node here doesn't make much sense. Considering `Nodeconfig` has node as a part of its struct. Not asking for a change here, this isn't a pattern you're introducing, just something i'm realizing now.

Reply:
@sebsoto thanks for catching this bad pattern. If we are doing this in other places, we should fix it sooner otherwise we will keep copying the same bad pattern.

Reply:
To be clear. It doesn't need to be fixed as part of this PR.